aws-datastore/src/main/java/com/amplifyframework/datastore/AmplifyDisposables.java

```diff
@@ -61,4 +71,47 @@ public boolean isDisposed() {
             }
         };
     }
+
+    /**
+     * This function that creates a {@link Consumer} which wraps the {@link ObservableEmitter#onError(Throwable)}
+     * to prevent it from calling observers that have already been disposed.
+     *
+     * <p>
+     * The scenario is that we have multiple event sources (1 AppSync subscription
+     * for each model+operation type combination) being consumed by a single downstream
+     * oberserver. Once the first subscription emits an error, the downstream subscriber
+     * is placed in a disposed state and will not receive any further notifications.
+     * In a situation such as loss of connectivity, it's innevitable that multiple subscriptions will fail.
+     * With that said, after the first failure, the other events sources (AppSync subscriptions)
+     * will attempt to invoke the downstream onError handler which then results in an
+     * {@link io.reactivex.exceptions.UndeliverableException} being thrown.
+     * </p>
+     *
+     * <p>
+     * This method takes a reference of the observable that represents the AppSync subscription,
+     * wraps it and returns a {@link Consumer} that is used as the onError parameter. The returned
+     * {@link Consumer} function will delegate the onError call to the downstream observers if it's
+     * still available, otherwise it logs a warning.
+     * </p>
+     *
+     * @param realEmitter The emitter which will have it's onError function proxied by the return
+     *                    value of this function.
+     * @param <T> The type of model handled by the emitter.
+     * @param <E> The type of exception for the onError consumer
+     * @return A {@link Consumer} that proxies the {@link ObservableEmitter#onError(Throwable)} call
+     * to the {@code realEmitter} if it's not disposed or logs a warning.
+     * @see <a href="https://github.com/aws-amplify/amplify-android/issues/541">GitHub issue #541</a>
+     *
+     */
+    @NonNull
+    public static <T extends Model, E extends AmplifyException> Consumer<E> onErrorConsumerWrapperFor(
```

Comment:
Is it an `AmplifyDisposable`s thing, really? It doesn't build a `Disposable`. If you only use it from the `SubscriptionProcessor`, maybe it should live as a static utility method on that class, instead?

Reply:
❤️ was granted for the wiki link! :-D